src/components/stepFour/utils.js

```diff
@@ -1136,3 +1138,39 @@ export const summaryFileContents = networkID => {
     }
   }
 }
+
+export const getStrategies = () => {
+  return ['Dutch', 'MintedCapped']
+}
+
+export const getPragmaVersion = async strategy => {
+  const strategiesAllowed = getStrategies()
+  if (!strategiesAllowed.includes(strategy)) {
+    throw new Error('Strategy not exist')
+  }
+  const contractFile = await (await fetch(`./contracts/${strategy}Proxy.sol`)).text()
+  const firstLine = contractFile.split('\n')[0]
+  return firstLine.match(/(?:\^0|\d*)\.(?:0|\d*)\.(?:0|\d*)/gi) || '0.4.24'
+}
+
+export const getOptimizationFlagByStrategy = strategy => {
+  const strategiesAllowed = getStrategies()
+  if (!strategiesAllowed.includes(strategy)) {
+    throw new Error('Strategy not exist')
+  }
+  const constants = require(`../../../public/metadata/${strategy}CrowdsaleTruffle.js`)
```

Comment:
`public` should be referenced here? I'm not sure about this, but if the _publish directory_ is pointing to `build/` the app may not have access to `/public`

Reply:
Hi @fernandomg , I make some changes taking into account the environment variables.
Best